src/connection_string.ts

```diff
@@ -203,7 +203,7 @@ function getUIntFromOptions(name: string, value: unknown): number {
 function* entriesFromString(value: string): Generator<[string, string]> {
   const keyValuePairs = value.split(',');
   for (const keyValue of keyValuePairs) {
-    const [key, value] = keyValue.split(':');
+    const [key, value] = keyValue.split(/:(.*)/);
```

Comment:
This change is to handle authMechanismProperties such as `TOKEN_AUDIENCE:api://foo` where we need the entire string after the first ":"

Reply:
… TIL that `.split()` inserts capture groups into the split result! :)

Reply:
Yeah that came in handy. :)